pkg/reconcile/reconcile_test.go

```diff
@@ -102,4 +114,75 @@ var _ = Describe("reconcile", func() {
 			Expect(err.Error()).To(Equal("nil terminal error"))
 		})
 	})
+
+	Describe("AsReconciler", func() {
+		var testenv *envtest.Environment
+		var testClient client.Client
+
+		BeforeEach(func() {
+			testenv = &envtest.Environment{}
+
+			cfg, err := testenv.Start()
+			Expect(err).NotTo(HaveOccurred())
+
+			testClient, err = client.New(cfg, client.Options{})
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			Expect(testenv.Stop()).NotTo(HaveOccurred())
+		})
+
+		Context("with an existing object", func() {
+			var key client.ObjectKey
+
+			BeforeEach(func() {
+				cm := &corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: "default",
+						Name:      "test",
+					},
+				}
+				key = client.ObjectKeyFromObject(cm)
+
+				err := testClient.Create(context.Background(), cm)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			It("should Get the object and call the ObjectReconciler", func() {
+				var actual *corev1.ConfigMap
+				reconciler := reconcile.AsReconciler[*corev1.ConfigMap](testClient, &mockObjectReconciler{
```

Comment:
I am very surprised this is needed - Could you try if upgrading to go 1.21 fixes that?

Reply:
This is indeed not needed in Go 1.21. I originally tested my PR locally against the Go 1.21 toolchain, but it appears that CI runs on Go 1.20. Are you suggesting upgrading the Go version in CI? If so, how do we do that?

Reply:
Looks like the Go 1.21 bump was done in the meantime. So I think you can use 1.21 now

Reply:
Rebase fixed it, thanks for the heads up.